### PR TITLE
sam rm: fixed command behavior when combine -d/-s/-f options + enabled '.' and '..' inputs

### LINKED ISSUE #570: sam rm -R
```
> sam rm --first-image 376 --last-image 596 -R .
SAM ERROR - You cannot remove folders "." or "..".
```

```
> sam rm --first-image 376 --last-image 596 -R
usage: sam rm [-h] [-a] [-d] [-s] [-f] [-e EXPRESSION] [-R]
              [--first-image FIRSTIMAGE] [--last-image LASTIMAGE]
              [--range RANGE RANGE] [--detect-negative] [-v VERBOSE]
              [--dry-run]
              inputs [inputs ...]
sam rm: error: too few arguments
```

### PR DESCRIPTION
**Pull request by cchampet**
Fix #570

### CODE REVIEW

File: applications/sam/sam_rm.py [hunk 1]
```diff
@@ -113,26 +114,32 @@ def _removeItems(self, items, args, detectionMethod, filters):
...
-                toRemove = False
+            toRemove = False
 
-            # sam-rm -s
-            if args.sequences and itemType != sequenceParser.eTypeSequence:
-                toRemove = False
+            # sam-rm default case: remove all items
+            if not args.directories and not args.files and not args.sequences:
+                toRemove = True
+            else:
```
Comment (fabiencastan) [inline applications/sam/sam_rm.py]: elif

File: applications/sam/sam_rm.py [hunk 2]
```diff
@@ -113,26 +114,32 @@ def _removeItems(self, items, args, detectionMethod, filters):
...
             # sam-rm -R
             if args.recursive and itemType == sequenceParser.eTypeFolder:
-                subFolder = os.path.join(item.getFolder(), item.getFilename())
-                self.logger.debug('Browse in sub folder"' + subFolder + '" with the following filters: ' + str(filters))
-                newItems = sequenceParser.browse(subFolder, detectionMethod, filters)
-                self._removeItems(newItems, args, detectionMethod, filters)
+                try:
+                    subFolder = os.path.join(item.getFolder(), item.getFilename())
+                    self.logger.debug('Browse in sub folder"' + subFolder + '" with the following filters: ' + str(filters))
+                    newItems = sequenceParser.browse(subFolder, detectionMethod, filters)
```
Comment (fabiencastan) [inline applications/sam/sam_rm.py:138]: It seems that this is not recursive.

Comment (cchampet) [inline applications/sam/sam_rm.py:138]: Because of the potential exception?

Comment (fabiencastan) [inline applications/sam/sam_rm.py:138]: `sequenceParser.browse` browse the input folder but not recursively. And I don't see any code around to make the recursion.

Comment (cchampet) [inline applications/sam/sam_rm.py:138]: The instruction just below: 

``` python
self._removeItems(newItems, args, detectionMethod, filters)
```

Comment (fabiencastan) [inline applications/sam/sam_rm.py:138]: Oh yes, you're right! I didn't see that it's the same function.

**Pull request merged**